include/stack-graphs.h

```diff
@@ -9,6 +9,40 @@
 #include <stdint.h>
 #include <stdlib.h>
 
+// The local_id of the singleton root node.
+#define SG_ROOT_NODE_ID 0
```

Comment:
Interesting that cbindgen makes a `#define` here and not a `const uint32_t`.

Reply:
`const uint32_t` allocates an actual variable with storage, so it's not safe in a header file, since you'd end up with multiple definitions of the same symbol.  You could use `static const uint32_t` to enforce that each definition is local to the file, or `extern const uint32` in the header and `const uint32_t` in a single `.c` file somewhere, but then you wouldn't be able to see that value of the variable in other files.

That said, `#defines` aren't easily visible from some FFI bindings, which is the whole point of this header file!  So that part is confusing. Maybe worth looking into as a separate issue.

Reply:
Looked into this and there's currently no way to override that cbindgen produces `#define` constants in C mode.  I think this is fine for now; it does mean that on the consuming side we'll have to define the constants by hand but I don't think that's a deal-breaker.